File: src/main/scala/com/pwootage/oc/js/duktape/DuktapeStatic.java
package com.pwootage.oc.js.duktape;

public class DuktapeStatic {
  private static boolean initialized = false;

  public static boolean isInitialized() {
    return initialized;
  }

  public static synchronized void initialize() {
    if (initialized) {
      return;
    }
    System.out.println("Initializing Duktape");
    initialized = true;
    native_init();
    System.out.println("SpiderMonkey Duktape");
  }

  native private static void native_init();
}

File: src/main/java/com/pwootage/oc/js/duktape/DuktapeStatic.kt
package com.pwootage.oc.js.duktape

object DuktapeStatic {
  var isInitialized = false
    private set

  fun initialize() {
    synchronized(this) {
      if (isInitialized) {
        return
      }
      println("Initializing Duktape")
      isInitialized = true
      native_init()
      println("SpiderMonkey Duktape")
    }
  }

  private external fun native_init()
}
